src/java/main/org/apache/zookeeper/server/quorum/QuorumCnxManager.java

```diff
@@ -647,11 +648,10 @@ public void run() {
                         numRetries = 0;
                     }
                 } catch (IOException e) {
-                    if (shutdown) {
```

Comment:
why we need to move code block **Line650-Line652** to code block **Line665-Line667** ?